browser-test/src/civiform_admin_program_statuses.test.ts

```diff
@@ -0,0 +1,189 @@
+      await dismissModal(pageObject)
+    })
+
+    it('fails to create status with an existing name', async () => {
+      await adminProgramStatuses.createStatus('Existing status')
+      await adminProgramStatuses.expectProgramManageStatusesPage(programName)
+      await adminProgramStatuses.createStatus('Existing status')
+      await adminProgramStatuses.expectProgramManageStatusesPage(programName)
+      await adminProgramStatuses.expectCreateStatusModalWithError(
+        'A status with name Existing status already exists',
+      )
+      await dismissModal(pageObject)
+    })
+  })
+
+  describe('edit existing statuses', () => {
+    const programName = 'test program edit statuses'
+    const firstStatusName = 'First status'
+    const secondStatusName = 'Second status'
+
+    beforeAll(async () => {
+      await adminPrograms.addProgram(programName)
+      await adminPrograms.gotoDraftProgramManageStatusesPage(programName)
+
+      // Create two existing statuses.
+      await adminProgramStatuses.createStatus(firstStatusName)
+      await adminProgramStatuses.expectProgramManageStatusesPage(programName)
+      await adminProgramStatuses.createStatus(secondStatusName)
+      await adminProgramStatuses.expectProgramManageStatusesPage(programName)
+      await adminProgramStatuses.expectStatusExists({
+        statusName: firstStatusName,
+        expectEmailExists: false,
+      })
+      await adminProgramStatuses.expectStatusExists({
+        statusName: secondStatusName,
+        expectEmailExists: false,
+      })
+    })
+
+    it('fails to edit status when providing an existing status name', async () => {
+      await adminProgramStatuses.editStatus(firstStatusName, {
+        editedStatusName: secondStatusName,
+      })
+      await adminProgramStatuses.expectProgramManageStatusesPage(programName)
+      await adminProgramStatuses.expectEditStatusModalWithError(
+        `A status with name ${secondStatusName} already exists`,
+      )
+      await dismissModal(pageObject)
+    })
+
+    it('fails to edit status with an empty name', async () => {
+      await adminProgramStatuses.editStatus(firstStatusName, {
+        editedStatusName: '',
+      })
+      await adminProgramStatuses.expectProgramManageStatusesPage(programName)
+      await adminProgramStatuses.expectEditStatusModalWithError(
+        'This field is required',
+      )
+      await dismissModal(pageObject)
+    })
+
+    it('edits an existing status name', async () => {
+      await adminProgramStatuses.editStatus(secondStatusName, {
+        editedStatusName: 'Updated status name',
+      })
+      await adminProgramStatuses.expectProgramManageStatusesPage(programName)
+      await adminProgramStatuses.expectStatusExists({
+        statusName: 'Updated status name',
+        expectEmailExists: false,
+      })
+      await adminProgramStatuses.expectStatusNotExists(secondStatusName)
+    })
+
+    it('edits an existing status and configures email', async () => {
```

Comment:
edit and delete email would be good cases too.

Reply:
Good call. Added to this test.